pkg/codegen/python/gen.go

```diff
@@ -2556,6 +2583,35 @@ func (mod *modContext) genType(w io.Writer, name, comment string, properties []*
 	}
 	fmt.Fprintf(w, "):\n")
 	mod.genTypeDocstring(w, comment, props)
+	if len(props) == 0 {
+		fmt.Fprintf(w, "        pass\n")
+	} else {
+		fmt.Fprintf(w, "        %s.__configure__(\n", name)
+		for _, prop := range props {
+			pname := PyName(prop.Name)
+			fmt.Fprintf(w, "            %[1]s=%[1]s,\n", pname)
+		}
+		fmt.Fprintf(w, "            _setter=lambda key, value: pulumi.set(__self__, key, value),\n")
+		fmt.Fprintf(w, "        )\n")
+	}
+
+	// Define method signature
+	fmt.Fprintf(w, "    @staticmethod\n")
+	fmt.Fprintf(w, "    def __configure__(*,")
+	for _, prop := range props {
+		pname := PyName(prop.Name)
+		ty := mod.typeString(prop.Type, input, false /*acceptMapping*/)
+		if prop.DefaultValue != nil {
+			ty = mod.typeString(codegen.OptionalType(prop), input, false /*acceptMapping*/)
+		}
+
+		var defaultValue string
+		if !prop.IsRequired() || prop.DefaultValue != nil {
+			defaultValue = " = None"
+		}
+		fmt.Fprintf(w, "\n             %s: %s%s,", pname, ty, defaultValue)
+	}
+	fmt.Fprintf(w, "\n             _setter=lambda key, value: ...):\n")
```

Comment:
Probably makes more sense to make this the _first_ argument to configure and not set any default, every use case we have of it we're going to pass a setting function.